risc0/build/src/docker.rs

```diff
@@ -205,6 +265,39 @@ fn compute_image_id(elf_path: &Path) -> Result<String> {
     Ok(image.compute_id().to_string())
 }
 
+/// Check if docker is running.
+fn check_docker(docker: impl AsRef<std::ffi::OsStr>, max_elapsed_time: Duration) -> Result<()> {
+    eprintln!("Checking if Docker is running...");
+    let backoff = backoff::ExponentialBackoffBuilder::new()
+        .with_max_elapsed_time(Some(max_elapsed_time))
+        .build();
+    let f = || -> Result<bool, backoff::Error<anyhow::Error>> {
+        Command::new(&docker)
+            .arg("version")
+            .stdout(std::process::Stdio::null())
+            .stderr(std::process::Stdio::null())
+            .status()
+            .context("Failed to run `docker version`")?
+            .success()
+            .then_some(true)
+            .ok_or(anyhow!("Docker engine is not running").into())
+    };
+
+    let result = backoff::retry(backoff, f);
+
+    match result {
+        Ok(true) => Ok(()),
+        _ => Err(anyhow!("Docker engine is not running")),
+    }
+}
+
+/// Ensures that docker is running.
+fn ensure_docker_is_running() -> Result<()> {
+    let docker = which("docker").context(DOCKER_MSG)?;
+    check_docker(docker, Duration::from_secs(5)).context(DOCKER_MSG)?;
```

Comment:
This doesn't feel very robust, why `5` seconds? My bigger worry is that we might mask underlying issues by trying to detect if docker is around. If we let a docker command fail on-demand, at least then the user can see the full message (and it might not be related to whether docker is running or not).

Reply:
Makes sense! 

I removed the docker checks ✅